cmd/thanos/store.go

```diff
@@ -36,6 +36,12 @@ func registerStore(m map[string]setupFunc, app *kingpin.Application, name string
 	chunkPoolSize := cmd.Flag("chunk-pool-size", "Maximum size of concurrently allocatable bytes for chunks.").
 		Default("2GB").Bytes()
 
+	maxSampleCount := cmd.Flag("grpc-sample-limit",
+		"Maximum amount of samples returned via a single Series call. 0 means no limit. NOTE: may overestimate the number of samples that would be needed to respond to a query.").
```

Comment:
What do you mean by this `NOTE`?

Reply:
That any given time-series which may appear in any given query may not be saved in the whole 2 hour block if it, for example, goes away because a scrape target goes down. How do you think we should improve this or remove it in general?

Reply:
So do you mean that we could have smaller chunks right? Yea, I would literally explain this:
```suggestion
		"Maximum amount of samples returned via a single Series call. 0 means no limit. NOTE: `NOTE: For efficiency we take 120 as number of samples in chunk, so the actualy number of samples might be lower, even though maxium could be hit").
```

Reply:
I think it's not great, but it's good for beginning (: